Runtime/Game/Requests/RemoteSessionRequest.cs

```diff
@@ -109,3 +126,175 @@ public class LootLockerStartRemoteSessionResponse : LootLockerSessionResponse
+        private static readonly Dictionary<Guid, LootLockerRemoteSessionProcess> _remoteSessionsProcesses =
+            new Dictionary<Guid, LootLockerRemoteSessionProcess>();
+
+        private static readonly Action<Guid> _continualPollingAction = (processGuid) =>
+            {
+                if (!_remoteSessionsProcesses.TryGetValue(processGuid, out var process))
+                {
+                    return;
+                }
+
+                if (process.ShouldCancel)
+                {
+                    LootLockerStartRemoteSessionResponse canceledResponse = new LootLockerStartRemoteSessionResponse
+                    {
+                        lease_status = LootLockerRemoteSessionLeaseStatus.Cancelled
+                    };
+                    process.ProcessCompletedCallbackAction?.Invoke(canceledResponse);
+                    _remoteSessionsProcesses.Remove(processGuid);
+                    return;
+                }
+
+                if (process.LeasingProcessTimeoutTime <= DateTime.UtcNow)
+                {
+                    LootLockerStartRemoteSessionResponse timedOutResponse = new LootLockerStartRemoteSessionResponse
+                    {
+                        lease_status = LootLockerRemoteSessionLeaseStatus.Timed_out
+                    };
+                    process.ProcessCompletedCallbackAction?.Invoke(timedOutResponse);
+                    _remoteSessionsProcesses.Remove(processGuid);
+                    return;
+                }
+
+                StartRemoteSession(process.LeaseCode, process.LeaseNonce, response =>
+                {
+                    if (!_remoteSessionsProcesses.TryGetValue(processGuid, out var processAfterStatusCheck))
+                    {
+                        return;
+                    }
+                    if (!response.success)
+                    {
+                        response.lease_status = LootLockerRemoteSessionLeaseStatus.Failed;
+                        processAfterStatusCheck.ProcessCompletedCallbackAction?.Invoke(response);
+                        _remoteSessionsProcesses.Remove(processGuid);
+                        return;
+                    }
+                    if (response.lease_status == LootLockerRemoteSessionLeaseStatus.Authorized)
+                    {
+                        processAfterStatusCheck.ProcessCompletedCallbackAction?.Invoke(response);
+                        _remoteSessionsProcesses.Remove(processGuid);
+                        return;
+                    }
+
+                    LootLockerRemoteSessionStatusPollingResponse pollingResponse =
+                        LootLockerResponse.Deserialize<LootLockerRemoteSessionStatusPollingResponse>(response);
+                    processAfterStatusCheck.UpdateCallbackAction?.Invoke(pollingResponse);
+                    processAfterStatusCheck.LastUpdatedAt = DateTime.UtcNow;
+
+                    Task.Delay(_leasingProcessPollingIntervalMilliseconds).ContinueWith(task =>
```

Comment:
This might not work in WebGL because of the use of Task. 
But it might also work, some things work with WebGL + tasks/async and some things do not work, so make sure to test it.
I would prefer if this would be made with a coroutine since that is fully "cross-platform" in Unity, where as Task/async might not work properly on some platforms (like WebGL for example).

But, since the possibility of doing this yourself with a coroutine exists I am okay with leaving this in but speciffying that it uses async and might not be supported on all platforms.